File: direct/test/java/ru/yandex/direct/core/testing/info/ContentPromotionBannerInfo.java
package ru.yandex.direct.core.testing.info;

import ru.yandex.direct.core.entity.banner.model.old.OldContentPromotionBanner;

public class ContentPromotionBannerInfo extends AbstractBannerInfo<OldContentPromotionBanner> {

    @Override
    public ContentPromotionBannerInfo withAdGroupInfo(AdGroupInfo adGroupInfo) {
        super.withAdGroupInfo(adGroupInfo);
        return this;
    }

    @Override
    public ContentPromotionBannerInfo withCampaignInfo(CampaignInfo campaignInfo) {
        super.withCampaignInfo(campaignInfo);
        return this;
    }

    @Override
    public ContentPromotionBannerInfo withBanner(OldContentPromotionBanner banner) {
        super.withBanner(banner);
        return this;
    }

    @Override
    public ContentPromotionBannerInfo withClientInfo(ClientInfo clientInfo) {
        super.withClientInfo(clientInfo);
        return this;
    }
}

File: direct/test/java/ru/yandex/direct/core/testing/info/banner/ContentPromotionBannerInfo.kt
package ru.yandex.direct.core.testing.info.banner

import ru.yandex.direct.core.entity.banner.model.ContentPromotionBanner
import ru.yandex.direct.core.entity.contentpromotion.model.ContentPromotionContent
import ru.yandex.direct.core.testing.info.adgroup.AdGroupInfo
import ru.yandex.direct.core.testing.info.adgroup.ContentPromotionAdGroupInfo

class ContentPromotionBannerInfo
    : BannerInfo<ContentPromotionBanner>() {

    lateinit var content: ContentPromotionContent

    val contentPromotionAdGroupInfo: ContentPromotionAdGroupInfo get() = adGroupInfo as ContentPromotionAdGroupInfo
    val contentId: Long get() = content.id

    fun isContentInitialized() = ::content.isInitialized

    override fun withAdGroupInfo(adGroupInfo: AdGroupInfo<*>): ContentPromotionBannerInfo {
        return super.withAdGroupInfo(adGroupInfo) as ContentPromotionBannerInfo
    }

    fun withContent(content: ContentPromotionContent) = apply {
        this.content = content
    }

    override fun withBanner(banner: ContentPromotionBanner): ContentPromotionBannerInfo {
        return super.withBanner(banner) as ContentPromotionBannerInfo
    }
}
